xarray/backends/netCDF4_.py

```diff
@@ -198,24 +201,40 @@ class NetCDF4DataStore(WritableCFDataStore, DataStorePickleMixin):
 
     This store supports NetCDF3, NetCDF4 and OpenDAP datasets.
     """
-    def __init__(self, filename, mode='r', format='NETCDF4', group=None,
+    def __init__(self, netcdf4_dataset, mode='r', group=None, writer=None,
+                 opener=None, autoclose=False):
+        self.ds = _get_netcdf4_group(netcdf4_dataset, group)
+
+        if autoclose and opener is None:
+            raise ValueError('autoclose requires an opener')
+
+        self._mode = mode
+        self._autoclose = autoclose
+        self._opener = opener
+        self._isopen = True
+        self.format = self.ds.data_model
+        self._filename = self.ds.filepath()
+        self.is_remote = is_remote_uri(self._filename)
+        super(NetCDF4DataStore, self).__init__(writer)
+
+    @classmethod
+    def open(cls, filename, mode='r', format='NETCDF4', group=None,
                  writer=None, clobber=True, diskless=False, persist=False,
                  autoclose=False):
+        import netCDF4 as nc4
+
         if format is None:
             format = 'NETCDF4'
-        opener = functools.partial(_open_netcdf4_group, filename, mode=mode,
+        opener = functools.partial(_open_netcdf4_group, filename,
                                    group=group, clobber=clobber,
                                    diskless=diskless, persist=persist,
                                    format=format)
-        self.ds = opener()
-        self._autoclose = autoclose
-        self._isopen = True
-        self.format = format
-        self.is_remote = is_remote_uri(filename)
-        self._filename = filename
-        self._mode = 'a' if mode == 'w' else mode
-        self._opener = functools.partial(opener, mode=self._mode)
-        super(NetCDF4DataStore, self).__init__(writer)
+
+        ds = nc4.Dataset(filename, mode=mode, format=format, clobber=clobber,
```

Comment:
So that was my attempt at not duplicating work needed in the opener, the constructor, and our new alternate constructor (`open`):
1. Open the filename and get a dataset
2. Find the group
3. Set the variables

If we drop the group (as discussed above) then all the constructor needs to do is turn off scale and mask. This would duplicate turning it off in opener--are you ok with the duplication of the "work" (not the code to do so)?